client/api_client.go

```diff
@@ -56,31 +56,32 @@ func NewAPIClient(c Config) *APIClient {
 func wrapRetryRequestDoer(doer rclient.RequestDoer) rclient.RequestDoerFunc {
 	return func(req *http.Request) (*http.Response, error) {
 		var response *http.Response
-		fn := func() (shouldRetry bool, err error) {
-			resp, err := doer.Do(req)
+		var err error
+		fn := func() (shouldRetry bool) {
+			var resp *http.Response
+			resp, err = doer.Do(req)
 			if err != nil {
-				return false, err
+				return false
 			}
 
 			if resp.StatusCode < 200 || resp.StatusCode > 299 {
-				serverError := readServerError(resp)
-				if serverError.Code == errors.EventualConsistencyError {
+				err = readServerError(resp)
+				if serverError, ok := err.(*errors.ServerError); ok && serverError.Code == errors.EventualConsistencyError {
```

Comment:
`readServerError` returns a `*errors.ServerError`; this type assertion is unnecessary